README.md

```diff
@@ -199,6 +199,55 @@ An error will be raised:
 
 `ActiveModel::ValidationError: Validation failed: Title can't be blank`
 
+#### Content Areas
+
+A component can be configured to have multiple 'content areas' which can be captured and later rendered in the component's view. For example:
+
+`app/components/modal_component.rb`:
+```ruby
+class ModalComponent < ActionView::Component::Base
+  validates :user, :header, :body, presence: true
+
+  with_content_areas :header
+
+  def initialize(user:)
+    @user = user
+  end
+end
+```
+
+`app/components/modal_component.html.erb`:
+```erb
+<div class="modal">
+  <div class="header"><%= header %></div>
+  <div class="body"><%= body %>"></div>
+</div>
+```
+
+We can render it in a view as:
+
+```erb
+<%= render(TestComponent, user: {name: 'Jane'}) do |component| %>
+  <% component.with(:header) do %>
+      Hello <%= user[:name] %>
+    <% end %>
+  <% component.with(:body) do %>
+    <p>Have a great day.</p>
+  <% end %>
+<% end %>
+```
+
+Which returns:
+
+```html
+<div class="modal">
+  <div class="header">Hello Jane</div>
+  <div class="body"><p>Have a great day.</p></div>
+</div>
+```
+
+Content for areas can be either as arguments to the render method or as the content of the block passed to the `with` method.
```

Comment:
```suggestion
Content for content areas can be passed as arguments to the render method or as named blocks passed to the `with` method.
```

Can you provide an example of this same demo component being rendered via the render args?

More broadly, I wonder if we should support content areas via render args at this point. Perhaps we could stick to just the `with` route for now?

Reply:
Sure I can provide the example. I think it's nice to be able to provide both. It 'just works' in that we don't have to do anything. If the Component has constructor args with the same name as content_areas then either version works. It will be clearer with an example.